ts/query.go

```diff
@@ -0,0 +1,411 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt@cockroachlabs.com)
+
+package ts
+
+import (
+	"container/heap"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/util"
+)
+
+// dataSpan is used to construct a normalized, monolithic view of a single time
+// series over an arbitrary time span. The actual data in a span may be stored
+// in multiple instances of InternalTimeSeriesData.
+type dataSpan struct {
+	startNanos  int64
+	sampleNanos int64
+	datas       []normalizedData
```

Comment:
nm...I found normalizedData below. But I think this might indicate that you should declare the most basic data structures first and then the ones which use them in that order...